### PR TITLE
IKC-228 Simple login to application

### PR DESCRIPTION
**Pull request by pbelke**

### CODE REVIEW

File: kouncil-frontend/apps/kouncil/src/app/login/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import {environment} from '../../environments/environment';
+
+@Component({
+  selector: 'app-login',
+  template: `
+    <app-kafka-navbar></app-kafka-navbar>
+
+    <div [ngClass]="backend === 'SERVER' ? 'kafka-desktop' : 'kafka-desktop-demo'">
+      <div class="main-login">
+        <form (ngSubmit)="login()" class="login-form">
```
Comment (dmejer) [inline kouncil-frontend/apps/kouncil/src/app/login/login.component.ts:16]: Can we have reactive form over here?
https://angular.io/guide/reactive-forms

File: kouncil-backend/src/main/java/com/consdata/kouncil/config/WebSecurityConfig.java [hunk 2]
```diff
@@ -1,20 +1,57 @@
...
+
+    @Bean
+    public AuthenticationManager authenticationManager(HttpSecurity http) throws Exception {
+        return http.getSharedObject(AuthenticationManagerBuilder.class)
+                .build();
+    }
+
+    @Bean
+    public InMemoryUserDetailsManager userDetailsService() {
+        UserDetails admin = User.withUsername("admin")
```
Comment (ynleborg) [inline kouncil-backend/src/main/java/com/consdata/kouncil/config/WebSecurityConfig.java:51]: Put info in README.md and docs/README.md like "default credentials are admin/admin)"

File: kouncil-frontend/apps/kouncil/src/app/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export class LoginComponent {
+
+  public backend: Backend = environment.backend;
+  username: string;
+  password: string;
+
+  constructor(private service: AuthService, private router: Router, private serverService: ServersService) {
+  }
+
+  login(): void {
```
Comment (dmejer) [inline kouncil-frontend/apps/kouncil/src/app/login/login.component.ts:43]: Try to make many small, dumb components instead smart.
In this component emit event to a parent component (some main screen component etc). Parent should decide what to do (invoke rest)

File: kouncil-backend/src/main/java/com/consdata/kouncil/config/WebSecurityConfig.java [hunk 4]
```diff
@@ -1,20 +1,57 @@
...
+public class WebSecurityConfig {
 
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
-        log.info("Configuring spring security");
-        http.csrf().disable().headers().contentSecurityPolicy("script-src 'self'");
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http
+                .csrf().disable()
```
Comment (dmejer) [inline kouncil-backend/src/main/java/com/consdata/kouncil/config/WebSecurityConfig.java:26]: csrf is supported by angular by default, you can easliy turn this on

File: kouncil-frontend/apps/kouncil/src/app/login/login.component.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      <div class="main-login">
+        <form (ngSubmit)="login()" class="login-form">
+          <input placeholder="Login" class="input" matInput type="text" [(ngModel)]="username"
+                 name="username">
+          <br>
+          <input placeholder="Password" class="input" matInput type="password"
+                 [(ngModel)]="password" name="password">
+          <br>
+          <button mat-button disableRipple class="action-button-white" type="submit">
+            Zaloguj
```
Comment (ynleborg) [inline kouncil-frontend/apps/kouncil/src/app/login/login.component.ts:24]: Login, not Zaloguj :)

File: kouncil-frontend/apps/kouncil/src/app/login/login.component.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+@Component({
+  selector: 'app-login',
+  template: `
+    <app-kafka-navbar></app-kafka-navbar>
+
+    <div [ngClass]="backend === 'SERVER' ? 'kafka-desktop' : 'kafka-desktop-demo'">
+      <div class="main-login">
+        <form (ngSubmit)="login()" class="login-form">
+          <input placeholder="Login" class="input" matInput type="text" [(ngModel)]="username"
```
Comment (ynleborg) [inline kouncil-frontend/apps/kouncil/src/app/login/login.component.ts:17]: put some text over the input like "Log in to your account"

File: kouncil-backend/src/main/java/com/consdata/kouncil/config/WebSecurityConfig.java [hunk 7]
```diff
@@ -1,20 +1,57 @@
...
+                    configuration.setAllowedMethods(List.of("*"));
+                    UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+                    source.registerCorsConfiguration("/**", configuration);
+                    return configuration;
+                })
+                .and()
+                .authorizeRequests()
+                .antMatchers("/api/login").permitAll()
+                .antMatchers("/api/info/version").permitAll()
+                .antMatchers("/**").authenticated();
```
Comment (dmejer) [inline kouncil-backend/src/main/java/com/consdata/kouncil/config/WebSecurityConfig.java:39]: what kind of authentication do you want to use in next step? jwt? db? mem? ldap?

File: kouncil-frontend/apps/kouncil/src/app/login/auth.service.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import {HttpClient} from '@angular/common/http';
+import {map} from 'rxjs/operators';
+import {Observable} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+
+  authenticated: boolean = false;
```
Comment (dmejer) [inline kouncil-frontend/apps/kouncil/src/app/login/auth.service.ts:11]: should be private BehaviourSubject - public observable to subscribe in components/guards

File: kouncil-frontend/apps/kouncil/src/app/login/login.component.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import {ChangeDetectionStrategy, Component} from '@angular/core';
+import {AuthService} from './auth.service';
+import {Router} from '@angular/router';
+import {configProviderFactory} from '../app.module';
+import {ServersService} from '@app/common-servers';
+import {Backend} from '@app/common-model';
+import {environment} from '../../environments/environment';
+
+@Component({
+  selector: 'app-login',
```
Comment (dmejer) [inline kouncil-frontend/apps/kouncil/src/app/login/login.component.ts:10]: Can you generate nx lib, move this component and similiar functionality over there?
You can call it common-login or feature-login etc
We have nx over here, please you libs :)

File: kouncil-frontend/apps/kouncil/src/app/login/login.component.scss [hunk 10]
```diff
@@ -0,0 +1,55 @@
+@import '../../styles/palette';
+
+.kafka-desktop {
+  height: calc(100% - 72px);
```
Comment (dmejer) [inline kouncil-frontend/apps/kouncil/src/app/login/login.component.scss:4]: this 72px should be constant somewhere

File: kouncil-frontend/apps/kouncil/src/app/navbar/navbar.component.ts [hunk 11]
```diff
@@ -14,11 +22,14 @@ import {ServersService} from '@app/common-servers';
       <img *ngIf="backendVersion$ | async as backendVersion"
            src="assets/kouncil-logo.png" alt="logo" class="kouncil-logo"
            matTooltip="{{backendVersion}}"/>
-      <a class="menu-button" mat-button disableRipple routerLinkActive="active" [routerLink]="['/topics']">Topics</a>
-      <a class="menu-button" mat-button disableRipple routerLinkActive="active" [routerLink]="['/brokers']">Brokers</a>
-      <a class="menu-button" mat-button disableRipple routerLinkActive="active" [routerLink]="['/consumer-groups']">Consumer
-        Groups</a>
-      <a class="menu-button" mat-button disableRipple routerLinkActive="active" [routerLink]="['/track']">Track</a>
+      <a class="menu-button" mat-button disableRipple routerLinkActive="active"
+         [routerLink]="['/topics']" *ngIf="authService.authenticated">Topics</a>
```
Comment (dmejer) [inline kouncil-frontend/apps/kouncil/src/app/navbar/navbar.component.ts:26]: async pipe is your friend. See progresbar service

**Review by dmejer — CHANGES_REQUESTED** [review]

File: kouncil-backend/src/main/java/com/consdata/kouncil/config/WebSecurityConfig.java (hunk 7 above)
Comment (pbelke) [reply kouncil-backend/src/main/java/com/consdata/kouncil/config/WebSecurityConfig.java]: Ldap

File: kouncil-frontend/apps/kouncil/src/app/login/login.component.scss (hunk 10 above)
Comment (pbelke) [reply kouncil-frontend/apps/kouncil/src/app/login/login.component.scss]: Moved to global css file

File: kouncil-frontend/libs/common-login/src/lib/login/user.ts [hunk 12]
```diff
@@ -0,0 +1,9 @@
+export class User {
```
Comment (dmejer) [inline kouncil-frontend/libs/common-login/src/lib/login/user.ts:1]: do you really need class?
see interface vs class in ts https://stackoverflow.com/a/51716899

File: kouncil-frontend/libs/common-login/src/lib/login/common-login.component.ts [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+    <div class="main-login">
+      <form (ngSubmit)="login()" class="login-form" [formGroup]="form">
+        <span class="login-info">Log in to your account</span>
+
+        <div #username class="login-field-container">
+          <div class="login-field-icon-container">
+            <mat-icon class="login-field-icon">person</mat-icon>
+          </div>
+          <div class="login-field-input-container">
+            <input placeholder="Login" class="input" matInput type="text" required
```
Comment (dmejer) [inline kouncil-frontend/libs/common-login/src/lib/login/common-login.component.ts:20]: https://stackoverflow.com/a/51716899
autocomplete="username" for login input
autocomplete="current-password" for  password input

File: kouncil-frontend/libs/common-login/src/lib/login/common-login.component.ts [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+  constructor(private fb: FormBuilder) {
+    this.form = this.fb.group({});
+    this.form.addControl('username', new FormControl('', Validators.required));
+    this.form.addControl('password', new FormControl('', Validators.required));
+  }
+
+  login(): void {
+    if (this.form.valid) {
+      this.loginUser.emit(new User(this.getControl('username').value, this.getControl('password').value));
```
Comment (dmejer) [inline kouncil-frontend/libs/common-login/src/lib/login/common-login.component.ts:60]: this.form.getRawValue()?

**Review by dmejer — CHANGES_REQUESTED** [review]

**Comment by dmejer** [comment]
gj

File: docs/README.md [hunk 15]
```diff
@@ -30,6 +30,9 @@ Images for Kouncil are hosted here: https://hub.docker.com/r/consdata/kouncil.
 
 For more advanced configuration consult the [Deployment](DEPLOYMENT.md#deployment) section.
 
+## Authentication
```
Comment (ynleborg) [inline docs/README.md:33]: There is also README.me in root folder just for github summary.

**Review by ynleborg — APPROVED** [review]

**Pull request merged**